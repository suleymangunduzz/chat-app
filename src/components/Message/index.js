import React from 'react';
import PropTypes from 'prop-types';

import { classnames, hoursFormatter, checkURLContainsImage } from '../../utils';

const Message = ({ side, body, time, timeType, userName, type = 'message' }) => {
    const contentClass = classnames({
        'message__content': true,
        'message__content--info': type === 'info',
        [`message__content--${side}`]: side && type !== 'info'
    });

    const titleClass = classnames({
        'message__content__title': true,
        [`message__content__title--${side}`]: side
    });

    const renderMessageContents = () => {
        return type === 'info' ? <div>{ body }</div>  : <>
            <div className={ titleClass }>{`${userName} ${hoursFormatter(time, timeType)}`}</div>
            { checkURLContainsImage(body) ? <img src={body} /> : <div className="message__content__text">{ body }</div> }
        </>;
    }

    return (
        <div className="message">
            <div className={ contentClass }>
                { renderMessageContents() }
            </div>
        </div>
    );
}

Message.propTypes = {
    side: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    userName: PropTypes.string.isRequired,
    time: PropTypes.object.isRequired,
    timeType: PropTypes.number.isRequired
}

export default Message;
