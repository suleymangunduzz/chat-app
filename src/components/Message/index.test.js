import React from 'react';
import renderer from 'react-test-renderer';
import Message from './index';

test('Message should be rendered on the right side', () => {
    const side = 'right';
    const body = 'Some text message';
    const time = { hours: 17, minutes: 40 };
    const timeType = 24;
    const userName = 'Süleyman GÜNDÜZ';
    const type = 'message';

    const component = renderer.create(<Message side={side} time={time} body={body} timeType={timeType} userName={userName} type={type}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Message should be rendered on the left side', () => {
    const side = 'left';
    const body = 'Some text message';
    const time = { hours: 17, minutes: 40 };
    const timeType = 24;
    const userName = 'Süleyman GÜNDÜZ';
    const type = 'message';

    const component = renderer.create(<Message side={side} time={time} body={body} timeType={timeType} userName={userName} type={type}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Message should be rendered with info type', () => {
    const side = 'left';
    const body = 'Some text message';
    const time = { hours: 17, minutes: 40 };
    const timeType = 24;
    const userName = 'Süleyman GÜNDÜZ';
    const type = 'info';

    const component = renderer.create(<Message side={side} time={time} body={body} timeType={timeType} userName={userName} type={type}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});