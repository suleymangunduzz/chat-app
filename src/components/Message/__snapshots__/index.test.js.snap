// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should be rendered on the left side 1`] = `
<div
  className="message"
>
  <div
    className="message__content message__content--left"
  >
    <div
      className="message__content__title message__content__title--left"
    >
      Süleyman GÜNDÜZ 17:40
    </div>
    <div
      className="message__content__text"
    >
      Some text message
    </div>
  </div>
</div>
`;

exports[`Message should be rendered on the right side 1`] = `
<div
  className="message"
>
  <div
    className="message__content message__content--right"
  >
    <div
      className="message__content__title message__content__title--right"
    >
      Süleyman GÜNDÜZ 17:40
    </div>
    <div
      className="message__content__text"
    >
      Some text message
    </div>
  </div>
</div>
`;

exports[`Message should be rendered with info type 1`] = `
<div
  className="message"
>
  <div
    className="message__content message__content--info"
  >
    <div>
      Some text message
    </div>
  </div>
</div>
`;
